file/files.go

```diff
@@ -0,0 +1,116 @@
+package file
+
+import (
+	"bytes"
+	"compress/zlib"
+	"crypto/sha1"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+)
+
+// Init sets up a file for dotfile to track.
+// Returns the alias for the newly tracked file.
+func Init(d *Data, filePath, altName string) (string, error) {
+	var (
+		alias string
+		err   error
+	)
+
+	if _, err = os.Stat(filePath); os.IsNotExist(err) {
+		return "", fmt.Errorf("\"%#v\" not found", filePath)
+	}
+
+	// Get the full path so that it can later turn it into a relative path.
+	fullPath, err := filepath.Abs(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	alias = altName
+	if altName == "" {
+		alias, err = pathToAlias(fullPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if err = d.setup(); err != nil {
+		return "", err
+	}
+
+	// Replace the full path with a relative path.
+	re := regexp.MustCompile(d.Home)
```

Comment:
This could panic, use the safe version

Reply:
Or, since this is just a static string, you could probably just use https://golang.org/pkg/strings/#Replace